### PR TITLE
Fix pydantic with nested annotated type annotations

### LINKED ISSUE #3107: With pydantic v2 a field marked as Optional[conint] throws an exception
## Description

I experienced this issue with `Optional[conint]`, but I suspect this will happen with any builtin pydantic field whose type ends up being an `Annotated` type.  I was able to reproduce this in the existing tests within `tests/experimental/pydantic/test_fields.py`, test method `test_conint`.  Just replace `field: pydantic.conint(lt=100)` with `field: Optional[pydantic.conint(lt=100)]`.  The exception is copied below in the additional context.

At least in my case I can work around this by just creating a validator for my field instead of using `conint`.  I only have one field that uses it.  However, it's definitely not ideal.

Example code below to see the error:

```python
class Model(pydantic.BaseModel):
    field: Optional[pydantic.conint(lt=100)]

@strawberry.experimental.pydantic.type(model=Model)
class Type:
    field: strawberry.auto
```

## System Information

 - Operating system: Mac OS
 - Strawberry version:  Need to use a new enough version for pydantic v2 support.

## Additional Context

I'm guessing we need to add some additional logic inside `replace_types_recursively` that handles the `Annotated` type more gracefully?  One thing that was interesting was when I was debugging, the `Annotated` type only comes in with `Optional`. If I just directly use `conint`, the type comes in as just an `int`.  I don't have a lot of experience with python type code so I was afraid to try and put in a fix.

```
tests/experimental/pydantic/test_fields.py:92 (test_conint)
@needs_pydantic_v2
    def test_conint():
        class Model(pydantic.BaseModel):
            field: Optional[pydantic.conint(lt=100)]
    
        @strawberry.experimental.pydantic.type(Model)
>       class Type:

test_fields.py:99: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../../strawberry/experimental/pydantic/object_type.py:183: in wrap
    all_model_fields: List[DataclassCreationFields] = [
../../../strawberry/experimental/pydantic/object_type.py:184: in <listcomp>
    _build_dataclass_creation_fields(
../../../strawberry/experimental/pydantic/object_type.py:67: in _build_dataclass_creation_fields
    get_type_for_field(field, is_input)
../../../strawberry/experimental/pydantic/object_type.py:49: in get_type_for_field
    replaced_type = replace_types_recursively(outer_type, is_input)
../../../strawberry/experimental/pydantic/fields.py:172: in replace_types_recursively
    converted = tuple(
../../../strawberry/experimental/pydantic/fields.py:173: in <genexpr>
    replace_types_recursively(t, is_input=is_input) for t in get_args(replaced_type)
../../../strawberry/experimental/pydantic/fields.py:180: in replace_types_recursively
    replaced_type = replaced_type.copy_with(converted)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = typing.Annotated[int, None, Interval(gt=None, ge=None, lt=100, le=None), None]
params = (<class 'int'>, None, Interval(gt=None, ge=None, lt=100, le=None), None)

    def copy_with(self, params):
>       assert len(params) == 1
E       AssertionError
```

### PR DESCRIPTION
**Pull request by gazorby**
## Description

Fix generation of inputs when using pydantic when model uses nested `Annotated` type annotations

## Types of Changes

- [ ] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Fixes #3107 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the CONTRIBUTING document.
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by botberry** [comment]
Thanks for adding the `RELEASE.md` file!



Here's a preview of the changelog:

---

Fix generation of input based on pydantic models using nested `Annotated` type annotations:

```python
import strawberry
from pydantic import BaseModel


class User(BaseModel):
    age: Optional[Annotated[int, "metadata"]]


@strawberry.experimental.pydantic.input(all_fields=True)
class UserInput:
    pass
```


---
Here's the preview release card for twitter:




Here's the tweet text: 
```
🆕 Release (next) is out! Thanks to Matthieu MN for the PR 👏

Get it here 👉 https://github.com/strawberry-graphql/strawberry/releases/tag/(next)
```

File: strawberry/experimental/pydantic/fields.py [hunk 1]
```diff
@@ -42,6 +42,15 @@
...
         raise
 
+try:
+    from typing import Annotated
+except ImportError:
+    # python < 3.9 does not have Annotated
+    if sys.version_info < (3, 9):
+        Annotated = ()
+    else:
+        raise
```
Comment (patrick91) [inline strawberry/experimental/pydantic/fields.py:52]: you can import it from typing extensions instead 😊

Comment (gazorby) [reply strawberry/experimental/pydantic/fields.py:52]: Right!

**Comment by gazorby** [comment]
@patrick91 Is there anything missing for this to be merged?

**Comment by patrick91** [comment]
@gazorby taking a look now! thanks for the ping 😊

**Review by patrick91 — APPROVED** [review]
I've added a minor note on the code, just something to check in future 😊

thanks!

**Pull request merged**